weld_rt/cpp/runtime.cpp

```diff
@@ -335,22 +338,26 @@ static void *thread_func(void *data) {
   pthread_mutex_unlock(&global_lock);
 
   int iters = 0;
-  // this work_loop call is needed to complete any work items that are initially on the queue
-  work_loop(td->thread_id, rd);
-  while (!rd->done) {
-    if (try_steal(td->thread_id, rd)) {
-      iters = 0;
-      work_loop(td->thread_id, rd);
-    } else {
-      // If this thread is stalling, periodically check for errors.
-      iters++;
-      if (iters > 1000000) {
-        if (rd->err != 0) {
-	  break;
-        }
+  try {
```

Comment:
According to this it shouldn't impose too much overhead, just increase binary size: https://stackoverflow.com/questions/16784601/does-try-catch-block-decrease-performance so we're probably okay